trismart_smartboard_integration/models/smartboard_connector.py

```diff
@@ -4,25 +4,27 @@ class SmartBoard(models.Model):
     _name = "smartboard.connector"
     _description = "SmartBoard Connector"
 
-    def param_check(self, sb_lead_id, x_api_key):
-        if not sb_lead_id:
-            return {"status": 400, "error": "Smartboard ID is null", "odoo_customer_id":False}
-        elif not x_api_key:
-            return {"status": 400, "error": "X API Key is null", "odoo_customer_id": False}
-        elif type(sb_lead_id) != int:
-            return {"status": 400, "error": "Smartboard ID is not an Integer", "odoo_customer_id": False}
-        elif type(x_api_key) != str:
-            return {"status": 400, "error": "X API Key is not an Integer", "odoo_customer_id": False}
-        else:
-            return 'Pass'
+    def param_check(self, input_dict):
+        res = []
+        if not input_dict['sb_lead_id']:
+            res = {"status": 400, "error": "Smartboard ID is null", "odoo_customer_id":False}
+        elif not input_dict['x_api_key']:
+            res = {"status": 400, "error": "X API Key is null", "odoo_customer_id": False}
+        elif type(input_dict['sb_lead_id']) != int:
+            res = {"status": 400, "error": "Smartboard ID is not an Integer", "odoo_customer_id": False}
+        elif type(input_dict['x_api_key']) != str:
+            res = {"status": 400, "error": "X API Key is not an Integer", "odoo_customer_id": False}
+        elif input_dict['project_template_id'] and type(input_dict['project_template_id']) == int:
+            res = "is_temp"
+        return res
 
     @api.model
     def create_project(self, sb_lead_id, x_api_key, project_template_id=None):
         try:
+            input_dict = {'sb_lead_id' : sb_lead_id, 'x_api_key' : x_api_key, 'project_template_id' : project_template_id}
             # Check input parameters:
-            x = self.param_check(sb_lead_id, x_api_key)
-            if x != 'Pass':
-                return x
+            if self.param_check(input_dict) and self.param_check(input_dict) != "is_temp":
+                return self.param_check(input_dict)
```

Comment:
Should the param_checked be called 3 times ? We should call only one time to know whether the input is valid or not? I suggest the function can return 2 values: Valid = True or False and error_resp = {"status": 400, "error": "....", "odoo_customer_id": False}.  If the valid is true, the error_resp can be empty dictionary.